components/brave_rewards/resources/extension/brave_rewards/background.ts

```diff
@@ -21,17 +19,23 @@ const iconOn = {
   }
 }
 
-// const iconOff = {
-//   path: {
-//     18: batIconOff18Url,
-//     36: batIconOff36Url,
-//     54: batIconOff54Url
-//   }
-// }
-
 chrome.browserAction.setBadgeBackgroundColor({ color: '#FB542B' })
 chrome.browserAction.setIcon(iconOn)
 
+chrome.tabs.query({
+  active: true,
+  currentWindow: true
+}, (tabs) => {
+  if (!tabs || !tabs.length) {
+    return
+  }
+
+  let tab = tabs[0]
+  if (tab && tab.id) {
+    rewardsPanelActions.init(tab)
```

Comment:
Why do we need an `init` stage now just for the active window's active tab? Do we depend on `onPublisherData` which happens whenever a tab is loaded? Either way, is there a reason to send `currentWindow: true` which is unreliable from the background script (which is across multiple windows). Presumably we should send multiple tabs and set the badge text/icon for all the active tabs in all windows? In fact, if we have the data, we could just set badge text for all the open tabs at the same time if we wanted to.

Reply:
true, we should set active tab for all windows. Fixed